incubator/java-spring-boot2/image/Dockerfile-stack

```diff
@@ -10,17 +17,49 @@ RUN  apt-get -qq update \
 
 COPY ./LICENSE /licenses/
 
-# OSGI core for version range processing
-COPY ./util /util
-RUN  /util/check_version build
+ENV APPSODY_STACK=${IMAGE_REGISTRY_ORG}/${STACK_ID}:${MAJOR_VERSION}.${MINOR_VERSION}
+LABEL appsody.stack="${APPSODY_STACK}"
 
 COPY ./project /project
 COPY ./config /config
+COPY ./templates /templates
+
 COPY ./mvn-stack-settings.xml /usr/share/maven/conf/settings.xml
 COPY ./mvn-stack-settings.xml /project/mvn-stack-settings.xml
 
+# Resolve templated values in pom and config files
+# For the parent pom:
+# Use the git repository slug as the maven group: appsody/stacks --> dev.appsody.stacks
+# Use the stack id as the maven artifact id: java-spring-boot2
+# Use the passed in version as the mvn artifact version
 WORKDIR /project
-RUN mvn --no-transfer-progress -B -N io.takari:maven:wrapper -Dmaven=$(mvn help:evaluate -Dexpression=maven.version -q -DforceStdout)
+RUN mvn --no-transfer-progress -B -N io.takari:maven:wrapper -Dmaven=$(mvn help:evaluate -Dexpression=maven.version -q -DforceStdout) \
+ && export MAVEN_GROUP_ID="dev.$(echo $GIT_ORG_REPO | sed 's|/|.|')" \
+ && export MAVEN_ARTIFACT_ID="${STACK_ID}" \
+ && export MAVEN_ARTIFACT_VERSION="${MAJOR_VERSION}.${MINOR_VERSION}.${PATCH_VERSION}" \
+ && export MAVEN_VERSION_RANGE=$(/project/java-spring-boot2-build.sh range "${MAJOR_VERSION}.${MINOR_VERSION}") \
+ && echo ${MAVEN_GROUP_ID}:${MAVEN_ARTIFACT_ID}:${MAVEN_ARTIFACT_VERSION} - ${MAVEN_VERSION_RANGE} \
+ && find . -name '*.xml' -exec \
+      sed -i -e "s|{{MAVEN_GROUP_ID}}|${MAVEN_GROUP_ID}|" \
+             -e "s|{{MAVEN_ARTIFACT_ID}}|${MAVEN_ARTIFACT_ID}|" \
+             -e "s|{{MAVEN_ARTIFACT_VERSION}}|${MAVEN_ARTIFACT_VERSION}|" \
+             -e "s|{{MAVEN_VERSION_RANGE}}|${MAVEN_VERSION_RANGE}|" {} \; \
+ && sed -i -e "s|{{APPSODY_STACK}}|${APPSODY_STACK}|" /project/Dockerfile
+
+# Package templates
+RUN if [ ${MAJOR_VERSION} -eq 0 ]; then \
+      export CONFIG_VERSION=${MAJOR_VERSION}.${MINOR_VERSION}; \
+    else \
+      export CONFIG_VERSION=${MAJOR_VERSION}; \
+    fi \
+ && for x in /templates/*; do \
+      cd $x; \
+      name=$(basename $x); \
+      echo "stack: ${IMAGE_REGISTRY_ORG}/${STACK_ID}:${CONFIG_VERSION}" > .appsody-config.yaml; \
+      tar -czf /templates/${name}.tar.gz --exclude-from=.gitignore .; \
```

Comment:
My personal preference would be for us to leave the full source of the templates in the stack image, leaving open the technique we use to install them to the init stage,

Reply:
I could easily leave the source for analysis.. but the tarball is needed by IDE and other tools. Extracting a directory of files is more work than extracting a tgz, it also has a greater impact on the release process.